### PR TITLE
Explain the behavior when Date constructor parameters under-/overflow

### LINKED ISSUE #4461: Issue with "Date() constructor": minor documentation improvement to `day` parameter
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

#### What information was incorrect, unhelpful, or incomplete?
`day` parameter

#### Specific section or headline?
Individual date and time component values

#### What did you expect to see?
It would be useful to point out that `0` references the last day of the previous month, e.g.:
```
day Optional
Integer value representing the day of the month. The default is `1`. `0` represents the last day of the previous month. 
```

#### Did you test this? If so, how?
Yes. In node:
```
>new Date(2021, 4, 0)
2021-04-30T05:00:00.000Z
```
(Where `4` for the month value is May.)

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/date/date`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/date/date/index.html
* Last commit: https://github.com/mdn/content/commit/528c5c997b8d5fce9565015d6a804f7998651322
* Document last modified: 2021-04-19T17:12:03.000Z

</details>

### LINKED ISSUE #6141: Issue with "Date() constructor": what happens if parameters are out of range?
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

#### What information was incorrect, unhelpful, or incomplete?
Incomplete concerning what happens (or should happen) if one or more values is outside of the valid range in the constructor.
For example, the month should be in the range from 0 to 11, inclusive.  What happens (or should happen) if you specify the month as 12 because it is December and you forgot to subtract 1?
Your example:  let birthday = new Date(1995, 11, 17)            // the month is 0-indexed
But what if we actually use:  let birthday = new Date(1995, 12, 17)            // Ooops !!!  the month is 0-indexed
#### Specific section or headline?
Examples

#### What did you expect to see?
How to test to see if the instance of the Date object is in fact valid or invalid, especially testing for invalid if parameters are out of the valid range.
Or, if it is not possible to test the Date object for validity, a statement that it is not possible to do so, perhaps with suggestions on how to test the Date object from outside the Date object.

#### Did you test this? If so, how?
With both Firefox and Safari, both running on Mac, I used isNaN(birthday) with an invalid value (month = 12) on my Date object, but isNaN gives false, indicating that the Date object seems to be valid.  I was hoping for true because the value for the month was invalid, which should make the entire Date object invalid.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/date/date`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/date/date/index.html
* Last commit: https://github.com/mdn/content/commit/8e4451e44af7128d2748bd37fa3be9afec5f18fe
* Document last modified: 2021-06-14T14:49:04.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Fix #4461 
Fix #6141

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/date/date/index.md [hunk 1]
```diff
@@ -43,9 +43,10 @@ new Date(year, monthIndex, day, hours, minutes, seconds, milliseconds)
 There are five basic forms for the `Date()` constructor:
 
 1. No parameters
-
-    When no parameters are provided, the newly-created `Date` object
-    represents the current date and time as of the time of instantiation.
+      
+    <p>
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/date/date/index.md:47]: I'd much rather we avoid these artifacts. I think this section might be more readable if we have H4 headings for the different forms, like:

```
#### No parameters

When no parameters are provided...
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
Very nice. I had a formatting suggestion only.

File: files/en-us/web/javascript/reference/global_objects/date/date/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/date/date/index.md]: Ref https://github.com/mdn/content/pull/9870

**Review by wbamberg — APPROVED** [review]
Thank you @Josh-Cena , looks much better and a very clear description of the extremely weird Date behavior.

**Comment by Josh-Cena** [comment]
Welcome 😄

**Comment by RobertRodes** [comment]
This is a much improved explanation. I'm sorry to have to report a minor typo: in the "Individual date and time component values" section, the final paragraph says "Similarly, if any parameter underflows, it "borrows" from the higher positions. For example, new Date(2020, 5, 0) will return May 31th, 2020." This needs to say "May 31st, 2020."

**Comment by Josh-Cena** [comment]
Oh, oops! Feel free to follow-up.

**Comment by teoli2003** [comment]
Yes, feel free to open a PR (with the Github UI it is quick, and we will happily review it). If you don't have time for it just say it here and we'll do it.

**Comment by RobertRodes** [comment]
I'm sorry, #4461 is the first time I opened an issue on github, so I'm a complete newbie. What is a PR, and where do I open it?

**Comment by Josh-Cena** [comment]
The [README](https://github.com/mdn/content/blob/main/README.md) has a pretty good summary already.

**Comment by teoli2003** [comment]
A PR, abbreviation of _pull request_, is a proposal for an actual change in the source. The proposal is _reviewed_, by a reviewer, then it is _merged_ and appears in production about 24 hours later.

There are different ways to create these PRs, but the easiest way for typo-like fixing is the following:
- Go to the source of the page on GitHub. In your case: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/date/date/index.md
- Click the edit button:
  Capture d’écran 2022-06-10 à 07 20 10
- You are in an editor, so… edit! 
- Go to the bottom and feel the info for the proposal:
  Capture d’écran 2022-06-10 à 07 21 29
- Finally, on the next screen, write a description of the change (this is the actual PR)

And that's it.

**Comment by RobertRodes** [comment]
Thank you! I believe I also had to click two different "Create Pull Request" buttons. Anyway, that's what I did. I hope I did it correctly.

**Pull request merged**